### PR TITLE
Implement new mailsending methods

### LINKED ISSUE #281: Mail rework
Endpoints to perform specific actions
- [ ] specific users
- [ ] teams
- [ ] users with ticket
- [ ] users without ticket
- [ ] users with pickup
- [ ] seatgroups
- [ ] manual

### PR DESCRIPTION
**Pull request by mgajanssen**
Fixes #281

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/utils/mail/MailRestController.java [hunk 1]
```diff
@@ -64,9 +76,60 @@ public MailRestController(MailService mailService, UserService userService, Team
...
+                .map(Ticket::getOwner)
+                .collect(Collectors.toList());
+
+        mailService.sendTemplateMailToAll(users, mailDTO);
+
+        return createResponseEntity(HttpStatus.OK, "Mail was successfully sent");
+    }
+
+    @RequestMapping(value = "/user/{seatgroup}", method = RequestMethod.POST)
+    ResponseEntity<?> sendMailToSeatGroup(@RequestParam String seatGroup, @Validated @RequestBody MailDTO mailDTO) {
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/utils/mail/MailRestController.java]: Why is this useful? Can't really think of a realistic case where we want to mail specific seatgroups

File: src/main/java/ch/wisv/areafiftylan/users/service/UserRepository.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.Collection;
```
Comment (mgajanssen) [inline src/main/java/ch/wisv/areafiftylan/users/service/UserRepository.java]: I think this collection does not have to be in here. It is never used.

File: src/main/java/ch/wisv/areafiftylan/utils/mail/MailRestController.java [hunk 3]
```diff
@@ -28,24 +32,32 @@
...
 @PreAuthorize("hasRole('ADMIN')")
 @RequestMapping("/mail")
 public class MailRestController {
 
+
+    private MailService mailService;
+    private UserService userService;
+    private TeamService teamService;
+    private TicketService ticketService;
+    private SeatService seatService;
```
Comment (mgajanssen) [inline src/main/java/ch/wisv/areafiftylan/utils/mail/MailRestController.java]: The seatservice can also be removed, the field is not used.

**Comment by mgajanssen** [comment]
It would also be very useful to be able to email all the users who have not yet reserved a seat.

**Comment by skamoen** [comment]
We can do two things. Either merge this and schedule a replacement, or not merge this and keep it as backup if we need this before a replacement is done... Both are fine with me, up to @mgajanssen

**Comment by mgajanssen** [comment]
I think it is good to keep this as a backup and not merge it. Lets find a better solution first.

**Comment by skamoen** [comment]
Closing, but keeping the branch.

**Pull request closed without merging**